queens/iterators/control_variates_iterator.py

```diff
@@ -0,0 +1,245 @@
+            num_samples (int): Number of samples on the cross-model estimator.
+            expectation_cv (float, opt): Expectation of the control variate. If the expectation is
+                                         None, it will be estimated via MC sampling.
+            num_samples_cv (int, opt): Number of samples to use for computing the expectation
+                                       of the control variate if this expectation is unknown.
+            use_optimal_num_samples (bool): Determines wether the iterator calculates and uses the
+                                            optimal number of samples to estimate the control
+                                            variate mean such that the variance of the control
+                                            variates estimator is minimized.
+            cost_model (float, opt): Cost of evaluating the model.
+            cost_cv (float, opt): Cost of evaluating the control variate.
+
+        Raises:
+            ValueError: If model is None.
+                        If control_variable is None.
+                        If neither the expectation nor the number of samples of the control
+                        variate are given and the optimal number of samples is not used.
+                        If the optimal number of samples are used and the costs of the models are
+                        not provided.
+        """
+        if model is None or control_variate is None:
+            raise ValueError("A model and a control variate have to be given!")
+
+        if expectation_cv is None and num_samples_cv is None and not use_optimal_num_samples:
+            raise ValueError(
+                "expectation_cv or num_samples_cv has to be given when not using "
+                "the optimal number of samples."
+            )
+
+        if use_optimal_num_samples and (cost_model is None or cost_cv is None):
+            raise ValueError(
+                "Model and control variate costs have to be given if you want to use the optimal "
+                "number of samples"
+            )
+
+        # Initialize parent iterator with no model.
+        super().__init__(model, parameters, global_settings)
+        self.control_variate = control_variate
+        self.seed = seed
+        self.num_samples = num_samples
+        self.samples = None
+        self.output = None
+        self.expectation_cv = expectation_cv
+        self.num_samples_cv = num_samples_cv
+        self.use_optimal_num_samples = use_optimal_num_samples
+        self.cost_model = cost_model
+        self.cost_cv = cost_cv
+        self.variance_cv_mean_estimator = 0
+
+    def pre_run(self):
+        """Draw samples for the core run."""
+        np.random.seed(self.seed)
+
+        self.samples = self.parameters.draw_samples(self.num_samples)
+
+    def core_run(self):
+        """Core run of iterator.
+
+        Computes the expectation estimate and its standard deviation.
```

Comment:
```suggestion
        Computes the cross-model estimator and its standard deviation.
```
Or control variates estimator?